units/provisioning_setup_host.go

```diff
@@ -898,3 +916,98 @@ func (j *setupHostJob) tryRequeue(ctx context.Context) {
+				h.Id, h.Provider)
+		}
+
+		// create the volume
+		volume, err := cloudMgr.CreateVolume(ctx, &host.Volume{
+			Size:             h.HomeVolumeGB,
+			AvailabilityZone: h.Zone,
+			CreatedBy:        h.StartedBy,
+			Type:             evergreen.DefaultEBSType,
+		})
+		if err != nil {
+			return errors.Wrapf(err, "can't create a new volume for host '%s'", h.Id)
+		}
+
+		// attach to the host
+		attachment := host.VolumeAttachment{VolumeID: volume.ID}
+		if err = cloudMgr.AttachVolume(ctx, h, &attachment); err != nil {
+			return errors.Wrapf(err, "can't attach volume '%s' to host '%s'", volume.ID, h.Id)
+		}
+
+		if err = h.SetHomeVolume(volume.ID, attachment.DeviceName); err != nil {
+			return errors.Wrapf(err, "can't set home volume for host '%s'", h.Id)
+		}
+	}
+
+	// run the distro's mount script
+	var err error
+	for i := 0; i < mountRetryLimit; i++ {
+		if err = errors.Wrap(mountLinuxVolume(ctx, env, h), "can't mount volume"); err == nil {
+			return nil
+		}
+		time.Sleep(mountSleepDuration)
+	}
+	return err
+}
+
+func mountLinuxVolume(ctx context.Context, env evergreen.Environment, h *host.Host) error {
+	client, err := h.JasperClient(ctx, env)
+	if err != nil {
+		return errors.Wrap(err, "can't get Jasper client")
+	}
+	defer func() {
+		grip.Warning(message.WrapError(client.CloseConnection(), message.Fields{
+			"message": "could not close connection to Jasper",
+			"host":    h.Id,
+			"distro":  h.Distro.Id,
+		}))
+	}()
+
+	deviceName := h.HomeVolumeDeviceName
+	if h.Distro.HomeVolumeDeviceName != "" {
+		deviceName = h.Distro.HomeVolumeDeviceName
+	}
+
+	// continue on umount mount error
+	exitCode, err := host.RunJasperCommandSynchronously(ctx, client, &options.Create{Args: []string{"sudo", "umount", "-f", fmt.Sprintf("/dev/%s", deviceName)}})
+	if err != nil && exitCode != umountMountErrorCode {
+		return errors.Wrap(err, "problem running umount command")
+	}
+
+	commands := []options.Create{
+		{Args: []string{"sudo", "/sbin/mkfs.xfs", "-f", fmt.Sprintf("/dev/%s", deviceName)}},
+		{Args: []string{"sudo", "mkdir", "-p", fmt.Sprintf("/%s", evergreen.HomeVolumeDir)}},
+		{Args: []string{"sudo", "mount", fmt.Sprintf("/dev/%s", deviceName), fmt.Sprintf("/%s", evergreen.HomeVolumeDir)}},
+		{Args: []string{"sudo", "ln", "-s", fmt.Sprintf("/%s", evergreen.HomeVolumeDir), fmt.Sprintf("%s/%s", h.Distro.HomeDir(), evergreen.HomeVolumeDir)}},
+		{Args: []string{"sudo", "chown", "-R", fmt.Sprintf("%s:%s", h.User, h.User), fmt.Sprintf("%s/%s", h.Distro.HomeDir(), evergreen.HomeVolumeDir)}},
+		{
+			Args:               []string{"sudo", "tee", "--append", "/etc/fstab"},
+			StandardInputBytes: []byte(fmt.Sprintf("/dev/%s /%s auto noatime 0 0", deviceName, evergreen.HomeVolumeDir)),
+		},
+	}
+	for _, command := range commands {
+		_, err := host.RunJasperCommandSynchronously(ctx, client, &command)
+		if err != nil {
+			return errors.Wrapf(err, "problem running command '%s'", strings.Join(command.Args, " "))
+		}
+	}
```

Comment:
this is... fine, but there are other options: 

1. the options.Create has an "OnSuccess" which allows you to chain commands off of each other (it's a little dodgy for shorter running commands.) 
2. if you use client.Command() rather than options create you can specify many subcommands, and let the jasper code handle all of the required looping.

I think I'm more in favor of 2.

Reply:
implemented option 2